### PR TITLE
Feat/atlas hacks

### PR DESCRIPTION
**Pull request by rnerella92**
Jira Ticket: [PXP-7690](https://ctds-planx.atlassian.net/browse/PXP-7690)

Setup Atlas login through Fence with OIDC, it will provide more flexible authentication flow, in contrast of current all-or-nothing approach.

### Bug Fixes
PXP-7690

### CODE REVIEW

File: .secrets.baseline [hunk 1]
```diff
@@ -56,167 +60,224 @@
...
-        "type": "Basic Auth Credentials",
-        "filename": "fence/config-default.yaml",
         "hashed_secret": "a94a8fe5ccb19ba61c4c0873d391e987982fbbd3",
         "is_verified": false,
-        "line_number": 31
+        "line_number": 31,
+        "type": "Basic Auth Credentials"
       },
       {
+<<<<<<< HEAD
```
Comment (paulineribeyre) [inline .secrets.baseline:85]: lots of conflicts in this file - you need to re-build the lock file

File: fence/blueprints/user.py [hunk 2]
```diff
@@ -24,6 +24,11 @@ def user_info():
         client_id = current_token["azp"]
     info = get_current_user_info()
     info["azp"] = client_id
+
+    # Atlas hack: Pretend this is actually a string.
+    # (sub is supposed to be string per the standard)
+    info["sub"] = str(info["sub"])
```
Comment (paulineribeyre) [inline fence/blueprints/user.py:30]: why is this an "atlas hack"?
and if we need to do this, let's do it in function `get_user_info` instead. You might need to update the API docs for this endpoint

File: fence/jwt/keys.py [hunk 3]
```diff
@@ -188,7 +188,7 @@ def public_key_to_jwk(self):
             except AttributeError:
                 # there is no need to decode values that are already strings
                 pass
-        jwk_dict.update({"use": "sig", "key_ops": "verify", "kid": self.kid})
+        jwk_dict.update({"use": "sig", "key_ops": ["verify"], "kid": self.kid})
```
Comment (paulineribeyre) [inline fence/jwt/keys.py:191]: could you explain this change? and again, the API docs for this endpoint might need an update

File: fence/jwt/token.py [hunk 4]
```diff
@@ -396,6 +396,13 @@ def generate_signed_access_token(
                 " running outside of flask application"
             )
 
+    audiences = []
+    if client_id:
+        audiences.append(client_id)
+    # append scopes for backwards compatibility
+    # eventual goal is to remove scopes from `aud`
+    audiences = audiences + scopes
```
Comment (paulineribeyre) [inline fence/jwt/token.py:404]: `audiences` is never used. could you explain this change?

File: .secrets.baseline (hunk 1 above)
Comment (rnerella92) [reply .secrets.baseline]: Updated the secrets file.

File: fence/blueprints/user.py (hunk 2 above)
Comment (m0nhawk) [reply fence/blueprints/user.py:30]: This is a non-complaint to the standard, per [OpenID](https://openid.net/specs/openid-connect-basic-1_0.html#rfc.section.2.2):
```
The sub value is a case sensitive string
```

The comment is redundant. :)

File: fence/jwt/keys.py (hunk 3 above)
Comment (m0nhawk) [reply fence/jwt/keys.py:191]: Again, this is non-complaint with the standard — [RFC7517](https://datatracker.ietf.org/doc/html/rfc7517#section-4.3):

```
Its value is an array of key operation values.
```

File: fence/jwt/token.py (hunk 4 above)
Comment (m0nhawk) [reply fence/jwt/token.py:404]: I'm not sure on this one. It looks like it's safe to remove.

**Review by m0nhawk — CHANGES_REQUESTED** [review]
@rnerella92 Some extra comments.

File: fence/blueprints/user.py (hunk 2 above)
Comment (Avantol13) [reply fence/blueprints/user.py:30]: I believe the suggestion was to do it here instead: https://github.com/uc-cdis/fence/blob/e2779111dce9a5db4de850a0ecc785373597b70b/fence/resources/user/__init__.py#L80 as this is the most low level support for constructing userinfo

File: fence/jwt/token.py (hunk 4 above)
Comment (Avantol13) [reply fence/jwt/token.py:404]: Let's remove it then

File: fence/blueprints/user.py (hunk 2 above)
Comment (m0nhawk) [reply fence/blueprints/user.py]: @rnerella92 and did you commit this change in the place Alex suggested?

**Comment by m0nhawk** [comment]
@Avantol13 Can you take a look again if everything is addressed?

**Review by Avantol13 — APPROVED** [review]

**Review by m0nhawk — APPROVED** [review]

**Pull request merged**